modules/integration/src/test/scala/scala/cli/integration/RunTestsDefault.scala

```diff
@@ -47,4 +47,21 @@ class RunTestsDefault extends RunTestDefinitions(scalaVersionOpt = None) {
       archLinuxTest()
     }
 
+  test("3.nightly") { // should run code using scala 3 nightly version
+    TestInputs(os.rel / "sample.sc" -> """println("Hello World")""").fromRoot {
```

Comment:
NIT it'd be great to verify the used version here as well (similar to `2.12.nightly` and `2.13.nightly` tests), but I figure we don't have an option to depend on a nightly version of the compiler dynamically. We can fix it some other time, I guess.